pom.xml

```diff
@@ -17,9 +17,9 @@
     <module>gapic-generator-java-pom-parent</module>
     <module>api-common-java</module>
     <module>gax-java</module>
-    <module>gapic-generator-java</module>
     <module>java-common-protos</module>
     <module>java-iam</module>
+    <module>gapic-generator-java</module>
```

Comment:
Without this change, presubmit builds (e.g. [build(11)](https://github.com/googleapis/gapic-generator-java/actions/runs/4564973991/jobs/8055440980?pr=1580)) fail on building from root.

```
Error:  Failed to execute goal on project gapic-generator-java: Could not resolve dependencies for project com.google.api:gapic-generator-java:jar:2.16.1-SNAPSHOT: 
Could not find artifact com.google.api.grpc:proto-google-iam-v1:jar:1.10.1-SNAPSHOT in google-maven-central-copy (https://maven-central.storage-download.googleapis.com/maven2) -> [Help 1]
```
Switching the submodules order resolved the issue for this PR, though I haven’t pinpointed which part of the changes led to existing build order breaking.

Leaving a note here for posterity, but will stop investigating for now since these code changes intend to be migrated away to a different repo.